File: oa-portal/problems/c14-cycle-finding/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> findNegativeCycle(int n, const vector<array<long long, 3>>& edges) {
    // write your code here

    return {};
}

int main() {
    int n, m;
    if (scanf("%d %d", &n, &m) != 2) return 0;
    vector<array<long long, 3>> edges(m);
    for (auto &e : edges) scanf("%lld %lld %lld", &e[0], &e[1], &e[2]);

    vector<int> cycle = findNegativeCycle(n, edges);
    if (cycle.empty()) { printf("NO\n"); return 0; }

    string out = "YES\n";
    for (size_t i = 0; i < cycle.size(); i++) {
        out += to_string(cycle[i]);
        out += (i + 1 == cycle.size() ? '\n' : ' ');
    }
    fwrite(out.data(), 1, out.size(), stdout);
    return 0;
}

File: oa-portal/problems/c14-cycle-finding/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    static int[] findNegativeCycle(int n, int[] ea, int[] eb, long[] ec) {
        // write your code here

        return new int[0];
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt(), m = in.nextInt();
        int[] ea = new int[m], eb = new int[m];
        long[] ec = new long[m];
        for (int i = 0; i < m; i++) { ea[i] = in.nextInt(); eb[i] = in.nextInt(); ec[i] = in.nextLong(); }

        int[] cycle = findNegativeCycle(n, ea, eb, ec);
        if (cycle.length == 0) { System.out.println("NO"); return; }

        StringBuilder sb = new StringBuilder("YES\n");
        for (int i = 0; i < cycle.length; i++) sb.append(cycle[i]).append(i + 1 == cycle.length ? '\n' : ' ');
        System.out.print(sb);
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
